app/views/govuk_publishing_components/components/_layout_header.html.erb

```diff
@@ -24,13 +25,15 @@
           </span>
         </span>
 
-        <span class="govuk-header__product-name">
-          <%= product_name %>
-        </span>
+        <% unless local_assigns[:hide_product_details] %>
```

Comment:
I'm aware this is not ready for review, but since it's open I would suggest we hide the wrapping span if no `product_name` is provided instead of introducing a new flag.